Gems/PhysX/Code/Source/JointComponent.cpp

```diff
@@ -140,22 +139,27 @@ namespace PhysX
 
         if (m_configuration.m_leadEntity.IsValid())
         {
-            Physics::RigidBodyRequestBus::EventResult(info.m_leadBody
-                , m_configuration.m_leadEntity
-                , &Physics::RigidBodyRequests::GetRigidBody);
-
-            if (!info.m_leadBody)
+            AzPhysics::SimulatedBodyComponentRequestsBus::EventResult(
+                info.m_leadBody, m_configuration.m_leadEntity, &AzPhysics::SimulatedBodyComponentRequests::GetSimulatedBody);
+
+            // Report a warning if there's no lead body or the body type is not one of the supported.
+            // In the future only body type validation will be needed 
+            if (!info.m_leadBody ||
+                !(info.m_leadBody->GetNativeType() == NativeTypeIdentifiers::RigidBody ||
+                 info.m_leadBody->GetNativeType() == NativeTypeIdentifiers::RigidBodyStatic))
             {
-                const AZStd::string entityWithoutBodyWarningMsg("Rigid body not found in lead entity associated with joint. Joint treated as constraint on global position.");
+                info.m_leadBody = nullptr;
+                const AZStd::string entityWithoutBodyWarningMsg(
+                    "Simulated body not found in lead entity associated with joint. Joint treated as constraint on global position.");
                 WarnInvalidJointSetup(m_configuration.m_leadEntity, entityWithoutBodyWarningMsg);
```

Comment:
As you mentioned before, think this should be info rather than a warning since it's an intended use case.